tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php

```diff
@@ -93,6 +93,34 @@ public function provideFixCases()
                 '<?php include (isHtml())? "1.html": "1.php";',
                 '<?php include (isHtml ())? "1.html": "1.php";',
             ],
+            [
+                '<?php
+$r = include($a . $b);
+$r = include($a . $b) ?><?php
+$r = $x ? include($a . $b) : $c;
+$r = array(include($a . $b) => $c);
+$r = foo(include($a . $b), $c);
+$r = print($a . $b) and foo();
```

Comment:
We have `$r = print($a . $b) and foo();`, but not `$r = print($a . $b) && foo();` - why is that? The list in `$logicalTokenKinds` looks far from complete.

Reply:
Because `print` has a [precedence](https://www.php.net/manual/en/language.operators.precedence.php) that is higher than `and` (and `xor` and `or`), but lower than the other expression-operators, including `&&` (whereas `include`/`require`[`_once`]'s precedence is the lowest, and `echo` isn't an expression so doesn't have a precedence).
I could add a test to show that `print ($a . $b) && foo()` is preserved, but there's already one for `print (2 + 3) / 4` (and one for `include ($a . $b) and foo()`), and static test cases will never be able to cover every possible combination anyway... WDYT?

`$logicalTokenKinds` is complete with all three `T_LOGICAL_`*, but would you prefer a name like "`$lowerPrecedenceThanPrintTokenKinds`"?

Reply:
I have added a test and inlined the variable (and reordered some lines) while updating my branch :slightly_smiling_face: